### PR TITLE
Add dogstatsd ARM build

### PR DESCRIPTION
**Pull request by kbogtob**
### What does this PR do?

Adds the dogstatsd ARM build to the Gitlab build pipeline. It uses the ARM runners and the ARM docker building images.

### Motivation

The dogstatsd build of the agent was not available for our customers.

### Additional Notes

* This PR does not include the building of the dogstatsd ARM docker image.
* I had to use the same hack as the ARM build of the agent.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -349,21 +387,21 @@ build_puppy_agent-deb_x64_arm:
...
-    - GOOS=linux GOARCH=arm inv -e agent.build --puppy
+    - inv -e agent.build --puppy
+    - $S3_CP_CMD $SRC_PATH/$AGENT_BINARIES_DIR/agent $S3_ARTIFACTS_URI/puppy/agent
 
-# build dogstatsd for deb-x64
-build_dogstatsd-deb_x64:
+# build puppy agent for ARM, to make sure the build is not broken because of build targets
+build_puppy_agent-deb_x64_arm:
   stage: binary_build
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:$DATADOG_AGENT_BUILDIMAGES
```
Comment (albertvaka) [inline .gitlab-ci.yml]: Can you fix this not actually building for ARM? Or are you going to do that in a different PR?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -349,21 +387,21 @@ build_puppy_agent-deb_x64_arm:
...
     - inv deps --verbose --dep-vendor-only --no-checks
   script:
-    - GOOS=linux GOARCH=arm inv -e agent.build --puppy
+    - inv -e agent.build --puppy
+    - $S3_CP_CMD $SRC_PATH/$AGENT_BINARIES_DIR/agent $S3_ARTIFACTS_URI/puppy/agent
 
-# build dogstatsd for deb-x64
-build_dogstatsd-deb_x64:
+# build puppy agent for ARM, to make sure the build is not broken because of build targets
+build_puppy_agent-deb_x64_arm:
```
Comment (albertvaka) [inline .gitlab-ci.yml]: I think the name should not have `x64` in it.

File: .gitlab-ci.yml (hunk 1 above)
Comment (kbogtob) [reply .gitlab-ci.yml]: I thought doing it in another PR

**Review by albertvaka — DISMISSED** [review]
LGTM if you fix the puppy build in a separate PR.

**Comment by kbogtob** [comment]
I'll add the release not in a few

File: .gitlab-ci.yml (hunk 2 above)
Comment (kbogtob) [reply .gitlab-ci.yml]: I kept it to be homogenous with the other ARM tasks and as the ARM building images have it in their names.

Comment (albertvaka) [reply .gitlab-ci.yml]: Yes, but that's a mistake :)

We should change those images to the open source ones, which don't have that typo anymore. See: https://github.com/DataDog/datadog-agent-buildimages/blob/master/.gitlab-ci.yml#L39

Comment (albertvaka) [reply .gitlab-ci.yml]: Probably a good idea to do it as part of this PR: https://github.com/DataDog/datadog-agent/pull/4654/files

Let's merge this one as it is.

**Review by albertvaka — APPROVED** [review]
The Github diff sucks but I think it is correct.
